### DISCUSSION TITLE
Auto-expiring records with time-to-live

### DISCUSSION THREAD

**Discussion by DaniDipp** [pocketbase/pocketbase#5183] [Q&A]
It would be great to be able to set a ttl on a record. When the time is up, the record would be automatically deleted and the realtime api notified.

**Comment by dschissler** [comment]
Any record that is deleted from within the same server process will be sending to the realtime API already.  The problem is in the detection of this, which seems incredibly need specific and with wildly different scalability and memory requirements.

If you're doing this yourself for a smaller dataset then you could search the table for the earliest date and then set a Golang timer to wake up the goroutine at that time.  Then you would delete that record, search for another record until none remained.  You'd have to either set up a poll wakeup timer, or to add a record insert or update hook that notifies the delete goroutine to start itself back up by releasing a sleep lock.

The problem to me seems that any implementation might be too inefficient once scaled at any level.

If you're not having many records then you could just do a dumb wakeup every 1s, 1m, 1h or 1d to do a hard scan on every record to see if it has expired.  You could probably get very far with that into the alpha development stage, and it seems very easy to make it.

Pocketbase has some crontab-like support.  You could probably just hook into that, which would save a lot of time on making the timing support.  I presume that the shutdown support on that would be baked in as well.

**Reply by whmeitzler** [reply]
I'd be surprised if you ended up with anything other than "wake up and scan the table for records with an update time older than now - timeout"

Might be worth adding an index on the updated field (or the expire-at field if that's how you want to do it)

If you do this in go, there's also an option to do the read query in a goroutine to keep it from blocking the server if it goes too long
